MobiusCore/Source/EffectHandlers/_EffectHandlerConnection.swift

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public typealias SideEffect = () -> Void
```

Comment:
To me, this is the 'effect handler'. Or perhaps, something like `EffectExecution`/`ExecuteEffect` if a verb is better (probably is?). It won't necessarily have a side effect, so I don't think this name is ideal. Names are hard.

Reply:
I don't agree.
I think `canHandle returns an optional SideEffect` is much clearer than `canHandle returns an optional EffectExecution`/`ExecuteEffect`

We have the notion of `Effect` in Mobius as a data-encoded version of a side-effect. The interpretation of an `Effect` should therefore naturally be `SideEffect`.